src/ephemeris.rs

```diff
@@ -399,6 +399,133 @@ impl Ephemeris {
     }
 }
 
+#[cfg(feature = "sbp-conversions")]
+mod sbp_error {
+    use crate::{signal::InvalidGnssSignal, time::InvalidGpsTime};
+    use std::error::Error;
+    use std::fmt;
+
+    #[derive(Debug, Copy, Clone, PartialOrd, PartialEq)]
+    pub enum EphemerisDecodeError {
+        InvalidTime(InvalidGpsTime),
+        InvalidSignal(InvalidGnssSignal),
+    }
+
+    impl fmt::Display for EphemerisDecodeError {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            match self {
+                EphemerisDecodeError::InvalidTime(time_err) => time_err.fmt(f),
+                EphemerisDecodeError::InvalidSignal(sig_err) => sig_err.fmt(f),
+            }
+        }
+    }
+
+    impl Error for EphemerisDecodeError {}
+
+    impl From<InvalidGpsTime> for EphemerisDecodeError {
+        fn from(other: InvalidGpsTime) -> EphemerisDecodeError {
+            EphemerisDecodeError::InvalidTime(other)
+        }
+    }
+
+    impl From<InvalidGnssSignal> for EphemerisDecodeError {
+        fn from(other: InvalidGnssSignal) -> EphemerisDecodeError {
+            EphemerisDecodeError::InvalidSignal(other)
+        }
+    }
+}
+
+#[cfg(feature = "sbp-conversions")]
+pub use sbp_error::EphemerisDecodeError;
+
+#[cfg(feature = "sbp-conversions")]
+impl TryFrom<sbp::messages::observation::MsgEphemerisGPS> for Ephemeris {
```

Comment:
I think you need to either fully qualify `TryFrom` or import it, it's not in the prelude like `From`. and maybe we should add `--all-features` to `cargo check` in the jenkinsfiles

Reply:
Good catch! I'll add those flags to the check and clippy commands and fix up any issues.